lib/IRGen/GenClangDecl.cpp

```diff
@@ -50,9 +50,15 @@ void IRGenModule::emitClangDecl(const clang::Decl *decl) {
   stack.push_back(decl);
 
   ClangDeclRefFinder refFinder([&](const clang::DeclRefExpr *DRE) {
-    const clang::ValueDecl *D = DRE->getDecl();
-    if (!D->hasLinkage() || D->isExternallyVisible())
-      return;
+    const clang::Decl *D = DRE->getDecl();
+    // Check that this is a file level decl and not inside a function.
```

Comment:
You should expand on this comment, since the loop also re-assigns D.  Something like:

  // If it's a member of a file-level decl, like a C++ static member variable,
  // we want to add the entire file-level declaration because Clang doesn't
  // expect to see members directly here.